### DISCUSSION TITLE
Can't get themes to work properly with iTerm 2

### DISCUSSION THREAD

**Discussion by jefflovejapan** [fish-shell/fish-shell#9422] [Q&A] [answered]
I'm not sure what I'm doing wrong but after several years of using Fish I still can't get themes to apply properly in iTerm 2 on macOS.

## What I'm doing

1. type `fish_config` at the prompt
2. choose a theme from the web config, clicking "Set Theme"
3. See a bunch of `$ set -U fish_color_normal...` scroll by in the terminal
4. Hit enter to exit the web config

## What I'm expecting

My iTerm window should look like the selected theme, either immediately, or for new windows/tabs, or after restarting iTerm.

## What's happening

My iTerm window stays the same (the web config says that Fish is unable to set the window's background color, so this is expected,) but certain types of characters take on the appearance of the selected theme. The output of `ls`, however, remains a dark blue that's barely legible on a dark background no matter which theme I choose.

Apologies but I know nothing about how these themes work under the hood or how to begin diagnosing what's going on here. I suspect that the Fish theme is interacting with the iTerm theme somehow but I have no idea how. Also curious how system-wide light- and dark-mode settings might be contributing to the issue.

**Comment by faho** [comment, marked as answer]
> The output of ls, however, remains a dark blue that's barely legible on a dark background no matter which theme I choose.

Fish's theme selection has nothing to do with ls' colors. How those are configured depends on the ls implementation, typically by setting the $LS_COLORS or $LSCOLORS variable to a specific, typically extremely cryptic, value. This is down to your ls, fish just tells it to enable colors.

You might want to try https://geoff.greer.fm/lscolors/, which will give you a "BSD LSCOLORS" value. Set it like

```fish
set -gx LSCOLORS exfxcxdxbxegedabagacad
```

in config.fish.

**Reply by jefflovejapan** [reply]
Awesome, thanks for the help!
